src/security/srv_acl.c

```diff
@@ -506,6 +506,76 @@ get_sec_capas_for_token(Auth__Token *token, struct ownership *ownership,
 	return rc;
 }
 
+static int
+get_sec_origin_for_token(Auth__Token *token, char **machine)
+{
+	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
+	int			machine_size = 0;
+	int			rc = 0;
+	char			*mtmp;
+	Auth__Sys		*authsys;
+
+
+	if (token == NULL || machine == NULL) {
+		D_ERROR("NULL input\n");
+		return -DER_INVAL;
+	}
+
+	rc = get_auth_sys_payload(token, &authsys);
+	if (rc != 0)
+		return rc;
+
+	if (authsys->machinename) {
+		machine_size = strnlen(authsys->machinename, MAXHOSTNAMELEN);
+	} else {
+		D_ERROR("Malformed AuthSys token missing machinename");
+		rc = -DER_INVAL;
+		goto out;
+	}
```

Comment:
IMO would be better to put this block first `if (authsys->machinename == NULL)` since it exits early. Then the other case needn't be indented

Reply:
Good idea. Swapped the logic around.

Reply:
Also in testing remembered that you can't compare a protobuf string field against NULL. It need to be compared to protobuf_c_empty_string.